src/vs/workbench/contrib/list/browser/tableColumnResizeQuickPick.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Table } from 'vs/base/browser/ui/table/tableWidget';
+import { DisposableStore } from 'vs/base/common/lifecycle';
+import Severity from 'vs/base/common/severity';
+import { localize } from 'vs/nls';
+import { IQuickInputService, IQuickPick, IQuickPickItem } from 'vs/platform/quickinput/common/quickInput';
+
+interface IColumnResizeQuickPickItem extends IQuickPickItem {
+	index: number;
+}
+
+export class TableColumnResizeQuickPick extends DisposableStore {
+	constructor(
+		private readonly _table: Table<any>,
+		@IQuickInputService private readonly _quickInputService: IQuickInputService,
+	) {
+		super();
+	}
+
+	async show(): Promise<void> {
+		const quickPick = this._quickInputService.createQuickPick<IColumnResizeQuickPickItem>();
+		const items: IColumnResizeQuickPickItem[] = [];
+		this._table.getColumnLabels().forEach((label, index) => {
+			if (label) {
+				items.push({ label, index });
+			}
+		});
+		quickPick.items = items;
+		quickPick.placeholder = localize('table.column.selection', "Select the column to resize, type to filter.");
+		quickPick.sortByLabel = false;
+		quickPick.canSelectMany = false;
+		quickPick.show();
+		this.add(quickPick.onDidAccept(() => {
+			const index = quickPick.selectedItems?.[0].index;
+			if (index !== undefined) {
+				this._selectColumn(quickPick, index);
+			}
+		}));
+		this.add(quickPick.onDidHide(() => {
+			quickPick.dispose();
+			this.dispose();
+		}));
+	}
+
+	private _selectColumn(quickPick: IQuickPick<IColumnResizeQuickPickItem>, index: number): void {
+		quickPick.items = [];
+		quickPick.placeholder = localize('table.column.resizeValue', "Please enter a width in percentage for the column.");
+		quickPick.show();
+		this.add(quickPick.onDidAccept(() => {
+			const percentage = Number.parseInt(quickPick.value);
+			if (!isNaN(percentage)) {
+				this._table.resizeColumn(index, percentage);
+				quickPick.hide();
+				return;
+			}
+		}));
+		this.add(quickPick.onDidChangeValue(() => this._validateColumnResizeValue(quickPick)));
+	}
```

Comment:
Do you think it would be a little easier if you just used the `pick` and `input` APIs?:

```ts
const items: IColumnResizeQuickPickItem[] = [];
this._table.getColumnLabels().forEach((label, index) => {
	if (label) {
		items.push({ label, index });
	}
 });
const column = await quickInputService.pick<IColumnResizeQuickPickItem>(
	items,
	{
		placeHolder: localize('table.column.selection', "Select the column to resize, type to filter."),
	}
);
if (!column) {
	return;
}
const percentage = await quickInputService.input({
	placeHolder: localize('table.column.resizeValue.placeHolder', "i.e. 20, 60, 100..."),
	prompt: localize('table.column.resizeValue.prompt', "Please enter a width in percentage for the '{0}' column.", column.label),
	validateInput: (input: string) => this._newValidateColumnResizeValue(input),
});
if (!percentage) {
	return;
}
this._table.resizeColumn(column.index, percentage);
```

where `_newValidateColumnResizeValue` is:
```ts
const percentage = Number.parseInt(input);
if (!Number.isInteger(percentage)) {
	return {
		content: localize('table.column.resizeValue.invalidType', "Please enter an integer."),
		severity: Severity.Error
	};
} else if (percentage < 0 || percentage > 100) {
	return {
		content: localize('table.column.resizeValue.invalidRange', "Please enter a number greater than 0 and less than or equal to 100."),
		severity: Severity.Error
	};
} else {
	return undefined;
}
```

Reply:
btw, I have not tested this code. I used the TS compiler in my brain (not very accurate)